### PR TITLE
Add Thai localization for Japan holidays

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

This PR adds Thai (`th`) l10n support for Japan.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/locale/th/LC_MESSAGES/JP.po [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Canada holidays th localization.
```
Comment (KJhellico) [inline holidays/locale/th/LC_MESSAGES/JP.po:1]:
```suggestion
# Japan holidays th localization.
```
In fact, that's all I can read. :))

Comment (PPsyrius) [reply holidays/locale/th/LC_MESSAGES/JP.po:1]: Oof, fixed now :)

Comment (arkid15r) [reply holidays/locale/th/LC_MESSAGES/JP.po]: Well, yeah! We trust you on this one :)

**Review by arkid15r — APPROVED** [review]
LGTM

**Pull request merged**